colcon_package_selection/package_selection/dependencies.py

```diff
@@ -103,13 +118,23 @@ def check_parameters(self, args, pkg_names):  # noqa: D102
             sys.exit('\n'.join(error_messages))
 
     def select_packages(self, args, decorators):  # noqa: D102
-        if args.packages_up_to:
-            select_pkgs = set(args.packages_up_to)
+        if (
+            args.packages_up_to is not None or
+            args.packages_up_to_regex is not None
```

Comment:
Nitpick: no need to check for `is not None` here - interpreting the arguments as booleans (as it was before) should be fine.

Reply:
Bad habit, sorry! Fixed as part of a73e2997549918b8b62042afa3a0982b8643fd53